test/pool-bin/helpers/MockBinPositionManager.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity ^0.8.19;
+
+import {CommonBase} from "forge-std/Base.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {IVault} from "pancake-v4-core/src/interfaces/IVault.sol";
+import {IBinPoolManager} from "pancake-v4-core/src/pool-bin/interfaces/IBinPoolManager.sol";
+import {IBinPositionManager} from "pancake-v4-periphery/src/pool-bin/interfaces/IBinPositionManager.sol";
+import {BinPositionManager} from "pancake-v4-periphery/src/pool-bin/BinPositionManager.sol";
+import {IAllowanceTransfer} from "permit2/src/interfaces/IAllowanceTransfer.sol";
+import {Planner, Plan} from "pancake-v4-periphery/src/libraries/Planner.sol";
+import {Actions} from "pancake-v4-periphery/src/libraries/Actions.sol";
+
+contract MockBinPositionManager is BinPositionManager, CommonBase {
+    using Planner for Plan;
+
+    constructor(IVault _vault, IBinPoolManager _binPoolManager, IAllowanceTransfer _permit2)
+        BinPositionManager(_vault, _binPoolManager, _permit2)
+    {}
+
+    function addLiquidity(IBinPositionManager.BinAddLiquidityParams calldata params)
```

Comment:
interesting.. i wonder if we should ask the hook dev to use this kind of contract 

context: they can't use universal to swap as universal router have re-entrancy guard